client/my-sites/people/main.jsx

```diff
@@ -33,9 +33,20 @@ class People extends React.Component {
 			case 'team':
 				return <TeamList site={ site } search={ search } />;
 			case 'followers':
-				return <FollowersList site={ site } />;
+				/* We're using the `key` prop here to make sure a fresh instance
+				   is mounted in case a user changes their site. That way we don't
+				   have to deal with resetting internal state. Same goes for email
+				   followers. */
+				return <FollowersList key={ `people-followers-${ site.ID }` } site={ site } />;
 			case 'email-followers':
-				return <FollowersList site={ site } search={ search } type="email" />;
+				return (
+					<FollowersList
+						key={ `people-email-followers-${ site.ID }` }
+						site={ site }
+						search={ search }
+						type="email"
```

Comment:
Uber nit: seeing a `type` prop outside an input file hurts my eyes, literally. 

Don't feel obligated to change that, not in this PR anyway. I just wanted to share this 😉